### PR TITLE
Mypy

### PR DESCRIPTION
**Pull request by ZidongLiu**
## Summary

- [x] mypy package update
- [x] mypy config
- [x] ci update
- [x] remove pruning related code
- [ ] Decide mypy rules to follow
    - [x] ignore-missing-imports = True
    - [ ] disallow_untyped_defs = True, will turn on later, this introduces another 150 errors
- [x] Fix broken files
- [x] vscode integration
- [x] command line
- [x] readme instruction for setting it up

### CODE REVIEW

**Comment by ZidongLiu** [comment]
fix 2 files
image

**Comment by ZidongLiu** [comment]
13 -> 12
image

**Comment by ZidongLiu** [comment]
65 errors -> 52 errors
image

**Comment by ZidongLiu** [comment]
decom_fl still works
image

52 -> 37 errors, 11 -> 9 files
image

**Comment by ZidongLiu** [comment]
37 -> 15, 9 -> 7 files
image

**Comment by ZidongLiu** [comment]
8 errors in 3 files.
image

still works
image

**Review by BichengYing — COMMENTED** [review]
Mypy looks great! some early comment.

File: cezo_fl/client.py [hunk 1]
```diff
@@ -41,17 +40,17 @@ def local_update(self, seeds: Sequence[int]) -> LocalUpdateResult:
         return NotImplemented
 
     @abc.abstractmethod
-    def reset_model(self) -> None:
+    def reset_model(self) -> None:  # type: ignore
```
[the remark this answers is not in the document]
Comment (ZidongLiu) [reply cezo_fl/client.py:43]: it's giving
image

**Comment by ZidongLiu** [comment]
@BichengYing @RogerRogerUSC need your help to resolve typing issue related to byzantine
image

**Review by BichengYing — APPROVED** [review]
LGTM, type-checking is great!

**Pull request merged**